wagtail/admin/tests/pages/test_edit_page.py

```diff
@@ -1871,12 +1871,25 @@ def test_edit_after_change_language_code_without_revisions(self):
             response, reverse("wagtailadmin_pages:edit", args=(self.child_page.id,))
         )
 
-    def test_page_edit_num_queries(self):
+    def test_page_edit_num_queries_as_superuser(self):
         # Warm up cache so that result is the same when running this test in isolation
         # as when running it within the full test suite
         self.client.get(reverse("wagtailadmin_pages:edit", args=(self.event_page.id,)))
 
-        with self.assertNumQueries(41):
+        with self.assertNumQueries(35):
+            self.client.get(
+                reverse("wagtailadmin_pages:edit", args=(self.event_page.id,))
+            )
+
+    def test_page_edit_num_queries_as_editor(self):
+        editor = self.create_user("editor", password="password")
+        editor.groups.add(Group.objects.get(name="Editors"))
+        self.login(username="editor")
+
+        # Warm up the cache as above.
+        self.client.get(reverse("wagtailadmin_pages:edit", args=(self.event_page.id,)))
+
+        with self.assertNumQueries(44):
```

Comment:
This surprisingly performs even better than #9594, one fewer query for superusers and three fewer queries for editors.